opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/OpenTracingShim.java

```diff
@@ -18,16 +18,34 @@
   private OpenTracingShim() {}
 
   /**
-   * Creates a {@code io.opentracing.Tracer} shim out of {@code OpenTelemetry.getTracerProvider()}
-   * and {@code OpenTelemetry.getPropagators()}.
+   * Creates a {@code io.opentracing.Tracer} shim out of {@code
+   * GlobalOpenTelemetry.getTracerProvider()} and {@code GlobalOpenTelemetry.getPropagators()}.
    *
    * @return a {@code io.opentracing.Tracer}.
    */
   public static io.opentracing.Tracer createTracerShim() {
-    return new TracerShim(
-        new TelemetryInfo(
-            getTracer(GlobalOpenTelemetry.getTracerProvider()),
-            GlobalOpenTelemetry.getPropagators()));
+    return createTracerShim(getTracer(GlobalOpenTelemetry.getTracerProvider()));
+  }
+
+  /**
+   * Creates a {@code io.opentracing.Tracer} shim using provided Tracer instance and {@code
+   * GlobalOpenTelemetry.getPropagators()}.
+   *
+   * @return a {@code io.opentracing.Tracer}.
+   */
+  public static io.opentracing.Tracer createTracerShim(Tracer tracer) {
+    return createTracerShim(tracer, OpenTracingPropagators.builder().build());
+  }
+
+  /**
+   * Creates a {@code io.opentracing.Tracer} shim using provided Tracer instance and {@code
+   * OpenTracingPropagators} instance.
+   *
+   * @return a {@code io.opentracing.Tracer}.
+   */
+  public static io.opentracing.Tracer createTracerShim(
```

Comment:
3 creational methods make me think we should be using a builder. any thoughts about that? It could definitely be done as a follow-up PR.

Reply:
maybe leave only two by removing `createTracerShim(OpenTelemetry openTelemetry)` ?
@carlosalberto ?

Reply:
Good call. Let's do that in a follow up. @malafeev would you mind creating an issue for that, so we don't forget, before merging this PR?

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/3084